torch/_inductor/codecache.py

```diff
@@ -1945,6 +1961,60 @@ def convert_arg(arg: Any) -> Any:
         return torch._C._aoti.unsafe_alloc_void_ptr_from_tensor(result)
 
 
+# Precompiled headers are persistent past program runtime, but associated with one
+# specific compiler version and set of flags.  We explicitly use default_cache_dir here
+# because these headers need to be global, rather than ignored by fresh_inductor_cache.
+_HEADER_DIR = os.path.join(default_cache_dir(), "precompiled_headers")
+_COMPILED_HEADERS: OrderedSet[str] = OrderedSet()
+
+
+def _precompile_header(
+    header: str, hashable_cmd_line: str, compiler: str, compile_command: dict[str, Any]
+) -> str:
+    compiler_version = get_compiler_version_info(compiler)
+    header_hash, header_full_path = write(
+        content=f"#include <{header}>\n",
+        extension="h",
+        extra=f"{hashable_cmd_line} {compiler_version} {torch.version.__version__}",
+        specified_dir=_HEADER_DIR,
+    )
+
+    if header_hash not in _COMPILED_HEADERS:
```

Comment:
With proper hashing, I think we can just check if the precompiled header file exists here.

Reply:
@desertfire Agreed that we could mostly do that at this point, but `_worker_compile_cpp` does that internally already and skips the compilation process if the file exists. I don't see any evidence locally that we lose much performance by leaving the file existence check in a single place here.